matrix_reminder_bot/callbacks.py

```diff
@@ -123,6 +130,11 @@ async def invite(self, room: MatrixRoom, event: InviteMemberEvent):
         """Callback for when an invite is received. Join the room specified in the invite"""
         logger.debug(f"Got invite to {room.room_id} from {event.sender}.")
 
+        # Don't respond to invites from disallowed users
+        if not is_allowed_user(event.sender):
+            logger.info(f"{event.sender} is not allowed, not responding to invite.")
```

Comment:
Technically, this would allow an attacker to invite-spam your log. The easy fix would be to make it a debug message like for the message callback above.

However, (assuming nio implements that, which I haven't checked,) I wonder if we should [ignore](https://spec.matrix.org/latest/client-server-api/#ignoring-users) the inviting user after rejecting them once to prevent that? This would reduce some "attack"-vectors by shifting the responsibility to the server and would encompass all 3 callbacks present here.

What makes this non-trivial is how to deal with users that have been (unintentionally) ignored, then added to the allowlist? I guess the bot would need to check and update the ignored users list on startup to make that work? I recognize this is a bit of a feature creep and the proposed implementation would work on a basic and probably-good-enough level. What do you think?

Reply:
I had a quick look at the documentation of nio and it seems like they did not implement *setting* global account data. The only way to modify `m.ignored_user_list` would be to create custom requests, perhaps using the `AsyncClient.send()` method(?). In general, I like your idea. But I feel like running custom requests against the homeserver is out of scope of this simple bot. :smile:

Reply:
So to sum it up... I'd call it an upstream issue for now and change the loglevel to INFO for this PR. :wink:
Maybe we can open up another issue so we remember it.